source/adapters/level_zero/kernel.cpp

```diff
@@ -241,21 +269,23 @@ UR_APIEXPORT ur_result_t UR_APICALL urEnqueueKernelLaunch(
     // Add the command to the command list, which implies submission.
     ZE2UR_CALL(zeCommandListAppendLaunchKernel,
                (CommandList->first, ZeKernel, &ZeThreadGroupDimensions, ZeEvent,
-                (*Event)->WaitList.Length, (*Event)->WaitList.ZeEventList));
+                WaitListLength, &TmpWaitList.ZeEventList[WaitEventOffset]));
```

Comment:
why events are still passed with in order queue?
we should have a way to pass null event waitlist with 0 length to utilize driver in order capabiltiies